prepare_pages.py

```diff
@@ -0,0 +1,386 @@
+    with open('/tmp/prepare_pages_saves.py', 'r') as f:
+        text = f.read()
+        new_files = eval(text)
+
+    print(':: Processing files...')
+    print('==> Making directories...')
+    print(' -> wiki-multi-html')
+    add_folder('wiki-multi-html', False)
+    for file in new_files.keys():
+        properties = new_files[file]
+        if properties["type"] == "folder":
+            print(' -> ' + file)
+            add_folder(file, False)
+    documentation_copied = False
+    for file in new_files.keys():
+        properties = new_files[file]
+
+        # Copy documentation folder from /tmp/doc/ to current directory
+        if properties["type"] == "documentation":
+            if not documentation_copied:
+                print("==> Copying Documentation folder...")
+                if exists("./doc/"):
+                    print(" -> Removing old documentation folder...")
+                    os.system(f"rm -rf doc")
+                print(" -> Copying new documentation files...")
+                os.system("cp -r /tmp/doc/ .")
+                documentation_copied = True  # Only copy the directory once
+            continue
+        if properties["type"] == "folder":
+            continue
+
+        new_name = properties["new_name"] if properties["new_name"] is not None else file
+        print(f'==> After processing {new_name}')
+
+        # If a file was converted with pandoc, it needs the stylesheet (in the header) and a footer.
+        if properties["convert_with_pandoc"]:
+            print("-> Applying Styles...")
+            with open(f"/tmp/{new_name}", 'r') as f:
+                text = f.read()
+            with open(new_name, 'w') as f:
+                f.write(header + text + footer)
+        else:
+            print(" -> Copying to current directory...")
+            os.system(f"cp \"/tmp/{new_name}\" .")
+
+        # Links need to be replaced from directory and markdown direct links (wiki.md) into website links (./wiki)
+        if properties["replace_links"]:
+            print(" -> Replacing links...")
+            for old in properties["replace_links"].keys():
+                new = properties["replace_links"][old]
+                if old != list(properties["replace_links"].keys())[-1]:
+                    print(f" |-> Replacing {old} with {new}...")
+                else:
+                    print(f" `-> Replacing {old} with {new}...")
+                # Need to use sed, as str.replace somehow misses some link changes?
+                os.system(f"sed -i 's#]({old}#]({new}#g' {new_name}")
+    print("==> Renaming 'wiki-multi-html' to 'wiki-multi'...")
+    os.system("mv './wiki-multi-html/' './wiki-multi'")
+    print(':: Done!')
+
+
+if __name__ == '__main__':
+    if len(sys.argv) == 1:
+        print('Error! Not enough arguments:')
+        print(f"Usage: '{sys.argv[0]}' <after|before>")
+        sys.exit(2)
+    if sys.argv[1] == 'before':
+        with open('.gh-pages.json', 'r') as config_file:
```

Comment:
Why don't you just json.load?

Reply:
(My) JSON (implementation) needs text to be imported, so just json.loads wouldn't work here.

Al I could do was remove the `config_file_contents` variable in cc231d524de9c785188bb5cce8850f5f9e47c20a.